### PR TITLE
Update SCREAM CreateUnitTest macro

### PR DESCRIPTION
**Pull request by jgfouca**
1) Adds same support for TEST_EXE that was added to the EKAT macro.
2) Use loops to avoid having to explicitly pass along every individual
option to the ekat macro.
3) Change resource_spread tests to take advantage of the new TEST_EXE
capability.
4) Update EKAT

### CODE REVIEW

File: components/scream/cmake/ScreamUtils.cmake [hunk 1]
```diff
@@ -44,90 +44,57 @@ function(CreateUnitTest target_name target_srcs scream_libs)
...
+  endforeach()
+
   if (SCREAM_MPI_EXTRA_ARGS)
     list(APPEND cut_options MPI_EXTRA_ARGS ${SCREAM_MPI_EXTRA_ARGS})
   endif ()
-  if (CreateUnitTest_COMPILER_DEFS)
-    list(APPEND cut_options COMPILER_DEFS ${CreateUnitTest_COMPILER_DEFS})
-  endif ()
-  if (CreateUnitTest_COMPILER_C_DEFS)
-    list(APPEND cut_options COMPILER_C_DEFS ${CreateUnitTest_COMPILER_C_DEFS})
```
Comment (jgfouca) [inline components/scream/cmake/ScreamUtils.cmake:97]: I noticed a lot of these hardcoded appends were not appending the same variable in the if check. For example, `CreateUnitTest_C_COMPILER_DEFS` was appended in multiple cases. I assume this was a copy/paste mistake and not something intentional?

Comment (bartgol) [reply components/scream/cmake/ScreamUtils.cmake:97]: Yeah, this looks a lot like a copy+paste error...

File: components/scream/cmake/ScreamUtils.cmake [hunk 2]
```diff
@@ -44,90 +44,57 @@ function(CreateUnitTest target_name target_srcs scream_libs)
...
-  if (CreateUnitTest_COMPILER_C_FLAGS)
-    list(APPEND cut_options COMPILER_C_FLAGS ${CreateUnitTest_C_COMPILER_DEFS})
-  endif ()
-  if (CreateUnitTest_COMPILER_CXX_FLAGS)
-    list(APPEND cut_options COMPILER_CXX_FLAGS ${CreateUnitTest_C_COMPILER_DEFS})
-  endif ()
+
   if (CreateUnitTest_COMPILER_F_FLAGS OR SCREAM_Fortran_FLAGS)
-    list(APPEND cut_options COMPILER_F_FLAGS ${CreateUnitTest_C_COMPILER_DEFS} ${SCREAM_Fortran_FLAGS})
+    list(APPEND cut_options COMPILER_F_FLAGS ${CreateUnitTest_F_COMPILER_DEFS} ${SCREAM_Fortran_FLAGS})
```
Comment (jgfouca) [inline components/scream/cmake/ScreamUtils.cmake:80]: Also, I was a bit confused why Fortran was the only language that got SCREAM_Fortran_FLAGS included.

**Review by bartgol — APPROVED** [review]
Nice cleanup by using loops!

**Comment by jgfouca** [comment]
Turned off automerge since I forgot to add `scripts` label.

**Review by bartgol — APPROVED** [review]

**Pull request merged**